integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-spark-service-bundle/nifi-spark-service/src/test/java/com/thinkbiganalytics/nifi/v2/spark/SparkJobserverServiceIT.java

```diff
@@ -0,0 +1,251 @@
+        }
+
+        @Override
+        public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+            // do nothing
+        }
+    });
+
+    /**
+     * Initialize instance variables.
+     */
+    @Before
+    public void setUp() throws Exception {
+
+        // Setup test runner
+        runner.addControllerService(SPARK_JOBSERVER_SERVICE_IDENTIFIER, sparkJobserverService);
+        runner.setProperty(SPARK_JOBSERVER_SERVICE_PROPERTY, SPARK_JOBSERVER_SERVICE_IDENTIFIER);
+        runner.setProperty(sparkJobserverService, SparkJobserverService.JOBSERVER_URL, sparkJobserverUrl);
+        runner.setProperty(sparkJobserverService, SparkJobserverService.SYNC_TIMEOUT, syncTimeout);
+        runner.enableControllerService(sparkJobserverService);
+    }
+
+    /**
+     * Verify creating Spark Context and then deleting the Context
+     */
+    @Test
+    @Ignore
+    public void testContextCreationAndDeletion() throws Exception {
+        // Test creating context
+        String contextName = "testContextCreationAndDeletion";
+
+        boolean created = sparkJobserverService.createContext(contextName, numExecutors, memPerNode, numCPUCores, sparkContextType, contextTimeout, async);
+        Assert.assertTrue(created);
+
+        boolean deleted = sparkJobserverService.deleteContext(contextName);
+        Assert.assertTrue(deleted);
+    }
+
+    /**
+     * Verify two threads attempting to create a Spark Context at once
+     */
+    @Test
+    @Ignore
+    public void testDuplicateContextCreation() throws Exception {
+        // Test creating context
+        String contextName = "testDuplicateContextCreation";
+
+        CreateSparkContext createSparkContext1 = new CreateSparkContext(contextName, numExecutors, memPerNode, numCPUCores, sparkContextType, contextTimeout, async, sparkJobserverService);
+        Thread thread1 = new Thread(createSparkContext1);
+
+        CreateSparkContext createSparkContext2 = new CreateSparkContext(contextName, numExecutors, memPerNode, numCPUCores, sparkContextType, contextTimeout, async, sparkJobserverService);
+        Thread thread2 = new Thread(createSparkContext2);
+
+        thread1.start();
+        thread2.start();
+
+        thread1.join();
+        thread2.join();
+
+        boolean contextExists = sparkJobserverService.checkIfContextExists(contextName);
+
+        Assert.assertTrue(contextExists);
+
+        sparkJobserverService.deleteContext(contextName);
+    }
+
+    /**
+     * Verify creating a Spark Context which times out and gets deleted
+     */
+    @Test
+    @Ignore
+    public void testContextTimeout() throws Exception {
+        // Test creating context
+        String contextName = "testContextTimeout";
+        int contextTimeout = 1;
+
+        boolean created = sparkJobserverService.createContext(contextName, numExecutors, memPerNode, numCPUCores, sparkContextType, contextTimeout, async);
+        Thread.sleep(15000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2925)